awsmp/_driver.py

```diff
@@ -22,34 +22,35 @@
 
 
 class AmiProduct:
-    def __init__(self, product_id: str):
+    def __init__(self, product_id: str, dry_run: bool = False):
         self.product_id: str = product_id
         self.offer_id = get_public_offer_id(product_id)
+        self._dry_run = dry_run
 
     @staticmethod
-    def create():
```

Comment:
this feels a bit clunky to me given we set the dry_run flag at the class level. do we have any usages that use the staticmethod that we should convert to using the instance?

Reply:
imo no, because except for create, everything relies on the `product_id` and/or `offer_id`. create is the exception in that neither exist at the time of this call